src/engine/enginesidechaincompressor.cpp

```diff
@@ -54,28 +54,28 @@ void EngineSideChainCompressor::processKey(const CSAMPLE* pIn, const int iBuffer
 
 double EngineSideChainCompressor::calculateCompressedGain(int frames) {
     if (m_bAboveThreshold) {
-        if (m_compressRatio < m_strength) {
-            m_compressRatio += m_attackPerFrame * frames;
-            if (m_compressRatio > m_strength) {
+        if (m_compressRatio > m_strength) {
+            m_compressRatio -= m_attackPerFrame * frames;
+            if (m_compressRatio < m_strength) {
                 // If we overshot, clamp.
                 m_compressRatio = m_strength;
             }
-        } else if (m_compressRatio > m_strength) {
+        } else if (m_compressRatio < m_strength) {
             // If the strength param was changed, we might be compressing too much.
-            m_compressRatio -= m_decayPerFrame * frames;
+            m_compressRatio += m_decayPerFrame * frames;
         }
     } else {
         if (m_compressRatio > 0) {
-            m_compressRatio -= m_decayPerFrame * frames;
-            if (m_compressRatio < 0) {
+            m_compressRatio += m_decayPerFrame * frames;
+            if (m_compressRatio > 1) {
                 // If we overshot, clamp.
-                m_compressRatio = 0;
+                m_compressRatio = 1;
             }
         } else if (m_compressRatio < 0) {
             // Complain loudly.
             qWarning() << "Programming error, below-zero compression detected.";
```

Comment:
VERIFY_OR_DEBUG_ASSERT and before the condition above.

Reply:
Hopefully I got your suggestion right [here](https://github.com/mixxxdj/mixxx/pull/2750/files#diff-e7a5ff7e9b435da4b512b10514dea4bcR68)
```
} else {
    VERIFY_OR_DEBUG_ASSERT(m_compressRatio >= 0) {
        qWarning() << "Programming error, below-zero compression detected.";
    }
    if (m_compressRatio < 1) {
        m_compressRatio += m_decayPerFrame * frames;
        if (m_compressRatio > 1) {
            // If we overshot, clamp.
            m_compressRatio = 1;
        }
    }
}
```
(sorry, I force-pushed)

Reply:
local engine tests are all passing now.